### ISSUE TITLE
LoggerParameterTypes inconsistent Nullability behaviour

### ISSUE THREAD

**Issue by Sin4ik02** [meziantou/Meziantou.Analyzer#889]
### Version of the Meziantou.Analyzer NuGet package

2.0.225

### Rule Identifier

MA0124

### Target Framework

net8.0

### C# Language version

12

### Description

Adding Nullable Logger Parameters of non-primitive types and strings results in errors and seems inconsistent in general (see steps to reproduce and temporary repository):

`Log parameter 'MyRecordNullable' must be of type 'global::System.Nullable<global::MA0124.Repro.MyRecord>' but is of type 'global::MA0124.Repro.MyRecord?'`

`Log parameter 'NameNullable' must be of type 'global::System.Nullable<global::System.String>' but is of type 'global::System.String?'`

Additionally, when [having non-nullable strings and records parameters in the LoggerParameterTypes file](https://github.com/Sin4ik02/MA0124.Repro/blob/4ac4e46614f79ef12f409185067cda4026eb3c57/MA0124.Repro/LoggerParameterTypes.txt#L7), but then pass a [Nullable variable in the actual log](https://github.com/Sin4ik02/MA0124.Repro/blob/4ac4e46614f79ef12f409185067cda4026eb3c57/MA0124.Repro/Logging.cs#L9-L10), this does not result in an error, although seems like it should.

### Reproduction Steps

Minimal code:

```c#
------------------------------------------
in your LoggerParameterTypes:
Name;System.String
NameNullable;System.Nullable{System.String}

Age;System.Int32
AgeNullable;System.Nullable{System.Int32}

MyRecord;MA0124.Repro.MyRecord
MyRecordNullable;System.Nullable{MA0124.Repro.MyRecord}

------------------------------------------------------

MyRecord record:

namespace MA0124.Repro;

public record MyRecord(string Name, int Age);

-----------------------------------------------------

Logging class:

namespace MA0124.Repro;

public static partial class Logging
{

    [LoggerMessage(1, LogLevel.Debug, "Record: {MyRecordNullable}, Age: {AgeNullable}, Name: {NameNullable}")]
    public static partial void LogNullableDetails(this ILogger logger, MyRecord? myRecordNullable, int? ageNullable, string? nameNullable);
    
    [LoggerMessage(1, LogLevel.Debug, "Record: {MyRecord}, Age: {Age}, Name: {Name}")]
    public static partial void LogDetails(this ILogger logger, MyRecord? myRecord, int? age, string? name);
}
```

### Other information

Temporary repro repo: https://github.com/Sin4ik02/MA0124.Repro

**Comment by meziantou** [comment]
The types in the configuration file are not valid.

[`Nullable<T>`](https://learn.microsoft.com/en-us/dotnet/api/system.nullable-1?view=net-9.0&WT.mc_id=DT-MVP-5003978) only supports `struct`. `string` and `MyRecord` are `class`, so the type `System.Nullable{System.String}` is not valid and doesn't match anything in your code. 
I've updated the error message in 7b801d93d0668b9fa6208360e829b9d3d6157ad5 to remove the nullable annotation from the type. The new message will be:

```
Log parameter 'NameNullable' must be of type 'global::System.Nullable<global::System.String>' but is of type 'global::System.String'
```

In you case, the config file should be:
```
Name;System.String
NameNullable;System.String

Age;System.Int32
AgeNullable;System.Nullable{System.Int32}

MyRecord;MA0124.Repro.MyRecord
MyRecordNullable;MA0124.Repro.MyRecord
```

**Comment by Sin4ik02** [comment]
Thanks for the quick response!
